supertokens/utils.go

```diff
@@ -277,7 +277,12 @@ func ErrorIfNoResponse(res http.ResponseWriter) error {
 }
 
 func MakeDefaultUserContextFromAPI(r *http.Request) UserContext {
-	return SetRequestInUserContextIfNotDefined(nil, r)
+	ctx := SetRequestInUserContextIfNotDefined(nil, r)
+
+	// Adding this here, as it the requestID needs to be added to the UserContext, once in the lifecycle of the request
+	(*ctx)["requestID"] = r.Context().Value(superTokensInstance.RequestIDKey)
```

Comment:
what if there is a framework in which the requestId is not part of the request context?

Reply:
Thanks for pointing out!

I have added the change to first look for the `X-Request-ID` HTTP `header` which is the unofficial standard (which the [latest HTTP RFC](https://www.rfc-editor.org/rfc/rfc9110.html#section-16.3.2.1-4) prohibits), if that's not found, only then we try to look into the `context`.